### PR TITLE
Add 44kHz WT and option for max voices

### PR DESCRIPTION
**Pull request by rsp4jack**
The 44kHz WT is found at <https://github.com/pedrolcl/sonivox/blob/0c7beb23c2da2f2b8d3aa7c7bf126b50c53e46f9/arm-wt-22k/jetcreator_lib_src/darwin-x86/wt_44khz.c>.

I think it weird that the 44k one has the same samples as the 22k one, only articulations and regions are different. How it works? :confused:

Tests are broken, and I do not know how to work on it.

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -98,9 +100,18 @@ list(APPEND SOURCES
 #arm-wt-22k/lib_src/eas_xmf.c
 #arm-wt-22k/lib_src/eas_xmfdata.c
 #arm-wt-22k/lib_src/jet.c
-  arm-wt-22k/lib_src/wt_22khz.c
 )
 
+if(USE_44KHZ)
+    list(APPEND SOURCES
+        arm-wt-22k/lib_src/wt_44khz.c
```
Comment (pedrolcl) [inline CMakeLists.txt]:
```suggestion
        arm-wt-22k/jetcreator_lib_src/darwin-x86/wt_44khz.c
```
I don't like the duplication of a big source file already existing on the same repository

**Review by pedrolcl — COMMENTED** [review]
> The 44kHz WT is found at https://github.com/pedrolcl/sonivox/blob/0c7beb23c2da2f2b8d3aa7c7bf126b50c53e46f9/arm-wt-22k/jetcreator_lib_src/darwin-x86/wt_44khz.c.

And should remain there. There is no need to make a copy.

> I think it weird that the 44k one has the same samples as the 22k one, only articulations and regions are different. How it works? 😕

`wt_22khz.c` has both 8 bits and 16 bits samples conditioned by the symbol `_8_BIT_SAMPLES`, while`wt_44khz.c` has only 8 bit samples.

> Tests are broken, and I do not know how to work on it.

The test suite uses the sample rate as an argument. It only needed a small change to fix the tests.

Anyway, this branch has lower quality results than the default, which uses 16 bits samples. The sample rate has less impact on the overall quality. Instead of using a different embedded soundfont, a better solution is to use an external DLS.

**Comment by pedrolcl** [comment]
You asked about how it works. The name `wt_44khz.c` is somewhat misleading. You interpreted that 44kHz was the sample rate of the included samples, but the each one of the 8 bit samples of this soundfont has different sample rate, and the only place where the number 44100 is used in this file is [here, at line number 14695](https://github.com/pedrolcl/sonivox/blob/0c7beb23c2da2f2b8d3aa7c7bf126b50c53e46f9/arm-wt-22k/jetcreator_lib_src/darwin-x86/wt_44khz.c#L14695). This value is compared with the output sample rate when loading the soundfont [here](https://github.com/pedrolcl/sonivox/blob/0c7beb23c2da2f2b8d3aa7c7bf126b50c53e46f9/arm-wt-22k/lib_src/eas_voicemgt.c#L3564).

Here is an alternate PR using the other soundfont `wt_22khz.c`, but also with audio output sample rate of 44.1 kHz:  #21

**Comment by rsp4jack** [comment]
Got it. Thanks for reviewing, #21 is nice, ~and I think it is okay to close this PR (#20).~

**Comment by rsp4jack** [comment]
[`2aed483` (#20)](https://github.com/pedrolcl/sonivox/pull/20/commits/2aed483b397d6e56e82c2341b4fede5346940b7d) removes `arm-wt-22k/lib_src/wt_44khz.c`, which is **different** from `arm-wt-22k/jetcreator_lib_src/darwin-x86/wt_44khz.c`. `arm-wt-22k/lib_src/wt_44khz.c` got 16 bits sample support. (I should mention it earlier :confused:)

**Comment by pedrolcl** [comment]
> [`2aed483` (#20)](https://github.com/pedrolcl/sonivox/pull/20/commits/2aed483b397d6e56e82c2341b4fede5346940b7d) removes `arm-wt-22k/lib_src/wt_44khz.c`, which is **different** from `arm-wt-22k/jetcreator_lib_src/darwin-x86/wt_44khz.c`. `arm-wt-22k/lib_src/wt_44khz.c` got 16 bits sample support. (I should mention it earlier 😕)

Then, you should re-add your file, commit and push it.

**Comment by pedrolcl** [comment]
This works with both sample rates 22050 and 44100. I still don't like having, in the same repository, two large source files that are mostly identical. What about unifying them into a single one? We could rename the unified source as: "wt_200k_g.c", which was the original file name according to one top comment.

By the way, you can find the original soundfont "wt_200k_G.sf2" in [this archive](https://musical-artifacts.com/artifacts/824).

**Comment by rsp4jack** [comment]
I just split the WTs and merged the same parts together. It should look better now.

**Review by pedrolcl — APPROVED** [review]

**Comment by pedrolcl** [comment]
Thanks for your work!

**Pull request merged**